enclave/src/polkadex_balance_storage.rs

```diff
@@ -227,14 +227,22 @@ pub fn lock_storage_and_reserve_balance(
 ) -> Result<(), GatewayError> {
     // Acquire lock on balance_storage
     let mutex = load_balance_storage()?;
-    let mut balance_storage: SgxMutexGuard<PolkadexBalanceStorage> = mutex.lock().unwrap();
-    let balance = balance_storage
-        .read_balance(token.clone(), main_acc.clone())
-        .expect("Unable to read balance from balance storage")
-        .clone();
-
+    let mut balance_storage: SgxMutexGuard<PolkadexBalanceStorage> = match mutex.lock() {
+        Ok(storage) => storage,
+        Err(_) => {
+            error!("Could not lock mutex of balance storage");
+            return Err( GatewayError::UnableToLock)
+        },
+    };
```

Comment:
Couldn't you improve on this, by using `map_error(...)?;` instead of the match ?

Reply:
The problem with map_err(...) is that there will be no error message in the worker. While debugging it helps aloooot when you get an error log right at the location where the error is created. I also think it's not perfect though...